nsxt/resource_nsxt_policy_tier0_inter_vrf_routing.go

```diff
@@ -0,0 +1,391 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var routeAdvertisementTypesValues = []string{
+	"TIER0_STATIC",
+	"TIER0_CONNECTED",
+	"TIER0_NAT",
+	"TIER0_DNS_FORWARDER_IP",
+	"TIER0_IPSEC_LOCAL_ENDPOINT",
+	"TIER1_STATIC",
+	"TIER1_CONNECTED",
+	"TIER1_LB_SNAT",
+	"TIER1_LB_VIP",
+	"TIER1_NAT",
+	"TIER1_DNS_FORWARDER_IP",
+	"TIER1_IPSEC_LOCAL_ENDPOINT",
+}
+
+func resourceNsxtPolicyTier0InterVRFRouting() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyTier0InterVRFRoutingCreate,
+		Read:   resourceNsxtPolicyTier0InterVRFRoutingRead,
+		Update: resourceNsxtPolicyTier0InterVRFRoutingUpdate,
+		Delete: resourceNsxtPolicyTier0InterVRFRoutingDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicyTier0InterVRFRoutingImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"gateway_path": getPolicyPathSchema(true, true, "Policy path for the Gateway"),
+			"bgp_route_leaking": {
+				Type:        schema.TypeList,
+				Description: "Import / export BGP routes",
+				Optional:    true,
+				MaxItems:    2,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"address_family": {
+							Type:         schema.TypeString,
+							Description:  "Address family type",
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"IPV4", "IPV6"}, false),
+						},
+						"in_filter": {
+							Type:        schema.TypeList,
+							Description: "route map path for IN direction",
+							Optional:    true,
+							MaxItems:    1,
```

Comment:
Is there a reason those lists with size 1 are still lists? If we're following the API that expects to grow in future, maybe we shouldn't code the `MaxItems` in the provider to prevent the requirement for future change.

Reply:
IDK why it's defined as it is - they're lists in the [spec](https://vdc-download.vmware.com/vmwb-repository/dcr-public/8e2a4b1a-659f-4be6-a4f4-7ca42b04cb84/9a2c1372-7174-40f0-89d2-27615bfd5d08/api_includes/types_BgpRouteLeaking.html). Do you think that we can remove `MaxItems` and let NSX enforce the limit?

Reply:
Otherwise I could flatten these attributes

Reply:
I'm in favor of letting NSX enforce the limit